### DISCUSSION TITLE
Applying multiple patches to one rom file in specific order❓

### DISCUSSION THREAD

**Discussion by fooness** [emmercm/igir#317] [Q&A] [answered]
I just read through the [patching](https://github.com/emmercm/igir/blob/main/docs/rom-patching.md) docs and wondered if it’s possible to apply multiple patches to one rom file, and—in addition—in a specific order; it seems there’s some patches (two examples below) that come with a multitude of different alternative and/or optional patches.

[A Link to the Past Redux](https://www.romhacking.net/hacks/2594/) via Romhacking.net

[Link's Awakening Redux](https://www.romhacking.net/hacks/4672/) via Romhacking.net

As for `.ips` patches we have to add the CRC32 checksum to the filename, maybe we could also add an `opt` (or something similar to indicate _additional_ or _optional_) keyword and incrementing index/number to the filename, like the examples below.

`Legend of Zelda, The - The Minish Cap (USA) abcebbb1.ips` # this would be the main patch _(already implemented)_
`Legend of Zelda, The - The Minish Cap (USA) abcebbb1_opt_1.ips`
`Legend of Zelda, The - The Minish Cap (USA) abcebbb1_opt_2.ips`
`Legend of Zelda, The - The Minish Cap (USA) abcebbb1_opt_3.ips`

What do you think? Or maybe there already is some feature/solution for achieving this?

**Comment by emmercm** [comment, marked as answer]
I'm wondering if users might be better off patching this way manually with Lunar IPS or similar.

I don't tend to make use of a lot of patches, personally. Have you found this multi-patch approach to be somewhat common? Usually, I come across simpler patches such as translations, or entire overhauls that are unlikely to be compatible with other patches such as the Pokemon Crystal or Fire Red reworks.

**Reply by fooness** [reply]
I tend to agree with you, and I myself also don’t do *that much* patching; in the past, when using RetroArch or similar, I rather often made use of the “soft patching” feature, which is not really a thing on MiSTer FPGA, and probably Analogue Pocket, too. That’s why I was rather happy to read about the patching feature in `igir`.

Maybe one day this feature will be wished for by many users; until then, the concept could exist as only a discussion. 😋
